src/java/htsjdk/variant/vcf/VCFHeaderLine.java

```diff
@@ -158,9 +158,13 @@ public static String toStringEncoding(Map<String, ? extends Object> keyValues) {
             builder.append("=");
             builder.append(entry.getValue().toString().contains(",") ||
                            entry.getValue().toString().contains(" ") ||
-                           entry.getKey().equals("Description") ? "\""+ entry.getValue() + "\"" : entry.getValue());
+                           entry.getKey().equals("Description") ? "\""+ escapeQuotes(entry.getValue().toString()) + "\"" : entry.getValue());
         }
         builder.append(">");
         return builder.toString();
     }
+
+    private static String escapeQuotes(final String value) {
+        return value.replaceAll("([^\\\\])\"", "$1\\\\\"");
```

Comment:
Could you add a comment here showing an example of what this replace does?  It's hard for me to read this regex.